pkg/hostpath/hostpath.go

```diff
@@ -221,13 +220,28 @@ func commonUpsertRanger(obj runtime.Object, podPath, filter string, key, value i
 	return nil
 }
 
+func commonOSRemove(dir string) error {
+	// we cannot do a os.RemoveAll on the mount point, so we remove all on each file system entity
+	// off of the potential mount point
+	return filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if info == nil {
+			return nil
+		}
+		// since we do not support mounting on existing content, a dir can only mean a share
+		// has been mounted as a separate dir in our share, so skip
+		if info.IsDir() {
```

Comment:
FYI, the e2e `TestTwoSharesSeparateButInheritedMountPathsRemoveTopPath` fails without this

Reply:
So if I understand this correctly, we can't mount shares onto existing directories in the container image?

Reply:
We can mount onto empty directories in the container image.

So like in the entitlements demo, since `/etc/pki/entitlement` is empty when you bring up the ubi8 image, that works.

Non-empty directories and what happens, based on my testing even prior to this change, is what the driver does not take on right now.  There is a lot of tl;dr around per mount tmpfs, facilitating selinux via the kubelet, etc.  If need be let's discuss that separately.

And fwiw, we can have two shares, where
- one has a mountPath of `/data`
- and the other has a mountPath of `/data/subdir`

this worked prior to this change, and still works with this change.

Reply:
all of which is touched upon in https://github.com/openshift/csi-driver-projected-resource#current-status-with-respect-to-the-kubernetes-volumemount-api